xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/display_macros.vm

```diff
@@ -17,9 +17,16 @@
 ## Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
 ## 02110-1301 USA, or see the FSF site: http://www.fsf.org.
 ## ---------------------------------------------------------------------------
-#macro (displayObjectProperty $propertyReference $displayMode)
+#macro (displayObjectProperty $propertyReference $displayMode $createIfMissing)
```

Comment:
The code looks good, but I'm not sure about the design.
My main concern is that `displayObjectProperty` is now making a bit more stuff than just displaying values, for a very specific UC. So sounds like we almost need a different macro for it (which is almost already the case since you added a new parameter), which would be called only for this UC. Now I don't know much if it would be doable to detect in display.vm if we are in the specific conditions of this UC to call the other macro.

Reply:
This is definitely possible, instead of setting the boolean parameter there could be an `#if` with a call to a different macro. Is your suggestion that this extra macro should just create the object and the property if missing (like `#createObjectAndPropertyIfMissing`) or should it replace the existing macro, i.e., basically be a variant of `#displayObjectProperty` like `#displayPotentiallyMissingObjectProperty`?

Reply:
> Is your suggestion that this extra macro should just create the object and the property if missing (like #createObjectAndPropertyIfMissing)

that sounds better to me yes.